src/ptvsd/adapter/debuggee.py

```diff
@@ -258,19 +269,60 @@ def _spawn_terminal(request, spawn_info):
     }
 
     try:
-        result = channels.Channels().ide().request("runInTerminal", body)
+        channels.Channels().ide().request("runInTerminal", body)
     except messaging.MessageHandlingError as exc:
         exc.propagate(request)
 
+    # Although "runInTerminal" response has "processId", it's optional, and in practice
+    # it is not used by VSCode: https://github.com/microsoft/vscode/issues/61640.
+    # Thus, we can only retrieve the PID via the "process" event, and only then we can
+    # start tracking it. Until then, nothing else to do.
+    pass
+
+
+def parse_pid(process_event):
+    assert process_event.is_event("process")
+
     global pid
-    pid = result("processId", int)
+    sys_pid = process_event("systemProcessId", int)
 
-    try:
-        ProcessTracker().track(pid, after_exit=lambda: _exited.set())
-    except Exception as exc:
-        # If we can't track it, we won't be able to detect if it exited or retrieve
-        # the exit code, so fail immediately.
-        request.cant_handle("Couldn't get debuggee process handle: {0}", str(exc))
+    if not _got_pid.is_set():
+        # Launched with "runInTerminal", so we had no way to get the PID before.
+        # Now that we do, start tracking it as usual.
+
+        def after_exit(code):
+            global exit_code
+            exit_code = code
+            _exited.set()
+
+        try:
+            pid = sys_pid
+            _got_pid.set()
+            ProcessTracker().track(pid, after_exit=after_exit)
+        except Exception as exc:
+            # If we can't track it, we won't be able to detect if it exited or retrieve
+            # the exit code, so fail immediately.
+            raise process_event.cant_handle(
+                "Couldn't get debuggee process handle: {0}", str(exc)
+            )
+
+    elif pid != sys_pid:
```

Comment:
This check will always fail if launch handler uses launcher.py to run user code.